saleor/graphql/warehouse/types.py

```diff
@@ -0,0 +1,96 @@
+from ..core.connection import CountableDjangoObjectType
+
+
+class WarehouseAddressInput(graphene.InputObjectType):
+    street_address_1 = graphene.String(description="Address.", required=True)
+    street_address_2 = graphene.String(description="Address.")
+    city = graphene.String(description="City.", required=True)
+    city_area = graphene.String(description="District.")
+    postal_code = graphene.String(description="Postal code.")
+    country = CountryCodeEnum(description="Country.", required=True)
+    country_area = graphene.String(description="State or province.")
+    phone = graphene.String(description="Phone number.")
+
+
+class WarehouseInput(graphene.InputObjectType):
+    name = graphene.String(description="Warehouse name.", required=True)
+    company_name = graphene.String(description="Company name.")
+    shipping_zones = graphene.List(
+        graphene.ID, description="Shipping zones supported by the warehouse."
+    )
+    email = graphene.String(description="The email address of the warehouse.")
+
+
+class WarehouseCreateInput(WarehouseInput):
+    address = WarehouseAddressInput(
+        description="Address of the warehouse.", required=True
+    )
+
+
+class WarehouseUpdateInput(WarehouseInput):
+    address = WarehouseAddressInput(
+        description="Address of the warehouse.", required=False
+    )
+
+
+class Warehouse(CountableDjangoObjectType):
+    class Meta:
+        description = "Represents warehouse."
+        model = models.Warehouse
+        interfaces = [graphene.relay.Node]
+        only_fields = [
+            "id",
+            "name",
+            "company_name",
+            "shipping_zones",
+            "address",
+            "email",
+        ]
+
+
+class StockInput(graphene.InputObjectType):
+    product_variant = graphene.ID(
+        required=True, description="Product variant associated with stock."
+    )
+    warehouse = graphene.ID(
+        required=True, description="Warehouse in which stock is located."
+    )
+    quantity = graphene.Int(description="Quantity of items available for sell.")
+
+
+class Stock(CountableDjangoObjectType):
+    stock_quantity = graphene.Int(
+        description="Quantity of a product available for sale.", required=True
+    )
+
+    quantity = graphene.Int(
+        required=True,
+        description="Quantity of a product in the warehouse's possession, "
+        "including the allocated stock that is waiting for shipment.",
+    )
+    quantity_allocated = graphene.Int(
+        required=True, description="Quantity allocated for orders"
+    )
+
+    class Meta:
+        description = "Represents stock."
+        model = models.Stock
+        interfaces = [graphene.relay.Node]
+        only_fields = ["warehouse", "product_variant", "quantity", "quantity_allocated"]
+
+    @staticmethod
+    def resolve_stock_quantity(root, *_args):
+        return get_available_quantity_for_customer(root)
+
+    @staticmethod
+    def resolve_quantity(root, *_args):
```

Comment:
This one and `resolve_quantity_allocated` should be wrapped with `@permission_required(ProductPermissions.MANAGE_PRODUCTS)`. These fields had these permissions in the `ProductVariant` type, where we deprecated them. We have to maintain the same permissions as we had there. Only `stockQuantity` is meant for display in public storefronts.